### PR TITLE
sci-geosciences/gmt version bump

### PR DESCRIPTION
**Pull request by uleysky**
Update ebuild to the version 4.5.17 and update the coastline database.
Closes: https://bugs.gentoo.org/512834

### CODE REVIEW

File: sci-geosciences/gmt/gmt-4.5.17.ebuild [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (Zlogene) [inline sci-geosciences/gmt/gmt-4.5.17.ebuild]: obsolete EAPI

File: sci-geosciences/gmt/gmt-4.5.17.ebuild [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+AUTOTOOLS_AUTORECONF=yes
+
+inherit autotools-utils multilib
```
Comment (Zlogene) [inline sci-geosciences/gmt/gmt-4.5.17.ebuild]: multilib is not needed in newer eapis for `$(get_libdir)`

File: sci-geosciences/gmt/gmt-4.5.17.ebuild [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+GSHHS="gshhs-2.2.0"
+
+DESCRIPTION="Powerful map generator"
+HOMEPAGE="http://gmt.soest.hawaii.edu/"
```
Comment (Zlogene) [inline sci-geosciences/gmt/gmt-4.5.17.ebuild]: https

File: sci-geosciences/gmt/gmt-4.5.17.ebuild [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		$(use_enable gmttria triangle)
+		)
+	autotools-utils_src_configure
+}
+
+src_install() {
+	autotools-utils_src_install install-all
+
+	# remove static libs
+	find "${ED}/usr/$(get_libdir)" -name '*.a' -exec rm -f {} +
```
Comment (Zlogene) [inline sci-geosciences/gmt/gmt-4.5.17.ebuild]: `|| die`

File: sci-geosciences/gmt/gmt-4.5.17.ebuild [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	"${FILESDIR}"/${PN}-4.5.9-no-strip.patch
+	"${FILESDIR}"/${PN}-4.5.6-respect-ldflags.patch
+	)
+
+AUTOTOOLS_IN_SOURCE_BUILD=1
+
+src_prepare() {
+	mv -f "${WORKDIR}/share/"* "${S}/share/" || die
+
+	tc-export AR RANLIB
```
Comment (Zlogene) [inline sci-geosciences/gmt/gmt-4.5.17.ebuild]: inherit toolchain-funcs?

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 2 above)
Comment (Zlogene) [inline sci-geosciences/gmt/gmt-4.5.17.ebuild]: `autotools-utils` is banned in modern EAPIs, do not use this

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 1 above)
Comment (uleysky) [reply sci-geosciences/gmt/gmt-4.5.17.ebuild]: Changed to EAPI=6

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 2 above)
Comment (uleysky) [reply sci-geosciences/gmt/gmt-4.5.17.ebuild]: Inherit only autotools.

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 3 above)
Comment (uleysky) [reply sci-geosciences/gmt/gmt-4.5.17.ebuild]: Fixed

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 4 above)
Comment (uleysky) [reply sci-geosciences/gmt/gmt-4.5.17.ebuild]: Fixed

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 5 above)
Comment (uleysky) [reply sci-geosciences/gmt/gmt-4.5.17.ebuild]: Removed tc-export. This is part of the old and outdated bugfix.

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 2 above)
Comment (uleysky) [reply sci-geosciences/gmt/gmt-4.5.17.ebuild]: Replaced on autotools

File: sci-geosciences/gmt/gmt-4.5.17.ebuild [hunk 6]
```diff
@@ -47,40 +45,37 @@ src_prepare() {
 	mkdir "${S}/share/coast"
```
Comment (Zlogene) [inline sci-geosciences/gmt/gmt-4.5.17.ebuild]: `|| die`

File: sci-geosciences/gmt/gmt-4.5.17.ebuild [hunk 7]
```diff
@@ -47,40 +45,37 @@ src_prepare() {
...
 	mv -f "${WORKDIR}/${GSHHG}/"*.nc "${S}/share/coast/" || die
 
-	tc-export AR RANLIB
-
-	autotools-utils_src_prepare
-
-	eautoreconf
+	default
```
Comment (Zlogene) [inline sci-geosciences/gmt/gmt-4.5.17.ebuild]: `default` comes first in function

File: sci-geosciences/gmt/gmt-4.5.17.ebuild [hunk 8]
```diff
@@ -47,40 +45,37 @@ src_prepare() {
...
+		$(use_enable postscript eps) \
+		$(use_enable mex) \
 		$(use_enable gmttria triangle)
-		)
-	autotools-utils_src_configure
 }
 
 src_install() {
-	autotools-utils_src_install install-all
+	emake DESTDIR="${D}" install-all || die
```
Comment (Zlogene) [inline sci-geosciences/gmt/gmt-4.5.17.ebuild]: no die

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 6 above)
Comment (uleysky) [reply sci-geosciences/gmt/gmt-4.5.17.ebuild]: Done

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 7 above)
Comment (uleysky) [reply sci-geosciences/gmt/gmt-4.5.17.ebuild]: Done

File: sci-geosciences/gmt/gmt-4.5.17.ebuild (hunk 8 above)
Comment (uleysky) [reply sci-geosciences/gmt/gmt-4.5.17.ebuild]: Done. As I understand it, "die" is not necessary, because the errors are handled inside the emake?

**Comment by uleysky** [comment]
I took into account all the comments of reviewers, but pull request is not closed. Need to do something else?

**Comment by Amynka** [comment]
Squash it to one commit. :)

**Comment by uleysky** [comment]
How? I can not merge commits that are already in the repository, and I can not specify another branch for this pull request.

**Comment by Zlogene** [comment]
@uleysky `git rebase -i origin/master` and `git push -f` then

**Comment by uleysky** [comment]
Thanks. Done.

**Pull request closed without merging**